allennlp/data/fields/sequence_field.py

```diff
@@ -1,7 +1,8 @@
-from allennlp.data.fields.field import Field
+from typing import Generic
+from allennlp.data.fields.field import Field, DataArray
 
 
-class SequenceField(Field):
+class SequenceField(Field[DataArray], Generic[DataArray]):
```

Comment:
this seems to be the way to have two levels of generic abstract classes. see e.g.

https://github.com/python/mypy/issues/2291